app/src/org/commcare/network/RemoteDataPullResponse.java

```diff
@@ -51,15 +61,12 @@ protected RemoteDataPullResponse(DataPullTask task,
      */
     public BitCache writeResponseToCache(Context c) throws IOException {
         BitCache cache = null;
-        try {
-            final long dataSizeGuess = ModernHttpRequester.getContentLength(response);
-
+        try (InputStream input = getInputStream()) {
```

Comment:
This line should be enough to resolve the issue, so let me know if the other changes doesn't make much sense.

Reply:
Got it. yeah I do think globalizing variables for no reason should be avoided. Also `HttpUtils.parseUserVisibleError` only gets called for a specific response code `406` in old code , so that behaviour must be retained.

Reply:
Cool! Will revert.